scripts/qa_crowbarsetup.sh

```diff
@@ -1134,8 +1134,17 @@ function onadmin_add_cloud_repo()
         done
     fi
 
-    echo -n "This cloud was installed on `cat ~/cloud` from: " | \
-        cat - /etc/cloudversion >> /etc/motd
+    cat >>/etc/motd <<EOMOTD
+This cloud was installed on `cat ~/cloud` from: `cat /etc/cloudversion`
+`if [[ $JENKINS_BUILD_URL ]] ; then
```

Comment:
Ugh no!  Don't nest complex code inside `cat`.  Instead:

    (
        echo "This cloud was installed on `cat ~/cloud` from: `cat /etc/cloudversion`"
        if [[ $JENKINS_BUILD_URL ]] ; then
            echo "Installed via Jenkins"
            echo "  created by the job:    $JENKINS_BUILD_URL"
            echo "  on the Jenkins worker: $JENKINS_NODE_NAME"
            echo "  using executor number: $JENKINS_EXECUTOR_NUMBER"
            echo "  using workspace path:  $JENKINS_WORKSPACE"
        fi
    ) >> /etc/motd